Assets/Scripts/BasketballPool.cs

```diff
@@ -0,0 +1,28 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class BasketballPool : ABaseEntity
+{
+    public int size = 5;
+    public GameObject basketball;
+    private List<GameObject> pool;
+    private int next = 0;
+
+    public override void BaseStart()
+    {
+        pool = new List<GameObject>();
+        for (int i = 0; i < size; i++)
+        {
+            GameObject obj = (GameObject)Instantiate(basketball);
+            obj.SetActive(false);
+            pool.Add(obj);
+        }
+    }
+
+    public GameObject getObject()
+    {
+        next++;
+        return pool[next %= size];

```

Comment:
Could we use regular if-statements for this? e.g.
```cs
next++;
if (next > size-1) {
    next = 0;
}
return pool[next];
```

The reason for this is that there is a potential error in the making. By using the modulo operator on `next`, we never decrement `next`. This means that if a player kept throwing basketballs infinitely (e.g. `next` increases infinitely), the game potentially could crash due to an out-of-bounds error for an int. Of course, this assumes that there is no time limit and that a player plays Hoops for a nearly infinite amount of time, but we should still avoid situations like this